### ISSUE TITLE
Issues getting two instances to sync

### ISSUE THREAD

**Issue by matthijsvp** [MISP/MISP#2939]
### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Support
| OS version (server)       | Debian
| OS version (client)       | 9
| PHP version               | 7.0.27-0+Deb9U1
| MISP version / git hash   | Local version: 2.4.87 /Remote version: 2.4.87
|Browser                   | N/A

### Expected behavior
Connection test works

### Actual behavior
Local version: 2.4.87
Remote version: 2.4.87
Status: OK
Compatiblity: Compatible
POST test: Invalid body

### Steps to reproduce the behavior
- We have two instances.
- Both instances have a sync user.
- When adding instance 2 as a server to instance 1, we can perform a connection test with the above results. The log on instance 2 shows that the request has been blackholed, but no clue why.
- When we click the "Explore" button on instance 1, we *do* get to see the events that have been created on instance 2.

Can you help with debugging? I have no clue what step to take next.

**Comment by iglocska** [comment]
Could you check the contents of the error log of the instance that produces the black hole? You can find it at `/var/www/MISP/app/tmp/logs/error.log`.

**Comment by matthijsvp** [comment]
```
2018-02-14 16:28:06 Error: [BadRequestException] The request has been black-holed
Request URL: /servers/postTest
Stack Trace:
#0 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Component/SecurityComponent.php(831): AppController->blackhole('auth')
#1 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Component/SecurityComponent.php(351): SecurityComponent->_callback(Object(ServersController), 'blackHole', Array)
#2 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Component/SecurityComponent.php(255): SecurityComponent->blackHole(Object(ServersController), 'auth', Object(AuthSecurityException))
#3 /var/www/MISP/app/Lib/cakephp/lib/Cake/Utility/ObjectCollection.php(128): SecurityComponent->startup(Object(ServersController))
#4 /var/www/MISP/app/Lib/cakephp/lib/Cake/Event/CakeEventManager.php(243): ObjectCollection->trigger('startup')
#5 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Controller.php(678): CakeEventManager->dispatch(Object(CakeEvent))
#6 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(189): Controller->startupProcess()
#7 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke(Object(ServersController), Object(CakeRequest))
#8 /var/www/MISP/app/webroot/index.php(92): Dispatcher->dispatch(Object(CakeRequest), Object(CakeResponse))
#9 {main}
```

**Comment by matthijsvp** [comment]
@iglocska ↑ could you help us a bit to the next step in debugging? Much appreciated!

**Comment by adulau** [comment]
Just wondering. What kind of network do you have between the two instances? Is there any intercepting devices (SSL/TLS proxy MiTM)?

**Comment by matthijsvp** [comment]
There is a firewall, but no intercepting. The instances are allowed to talk to each other on port 443. Funny thing is, When we click the "Explore" button on instance 1, we do get to see the events that have been created on instance 2.

**Comment by matthijsvp** [comment]
You got me to thinking though, could this be caused by invalid (or perhaps self-signed) certificates? I don't have access to the instances this week, but could check for that next week.

**Comment by adulau** [comment]
Not sure. Did you import the self-signed certificate chain as PEM? Let me know how this goes.

**Comment by matthijsvp** [comment]
Seems to have been fixed when updating from v2.4.90 to 2.4.93. Thanks for the help!
